client_restaurant/shared/src/commonMain/kotlin/domain/usecase/ManageMealUseCase.kt

```diff
@@ -13,7 +14,7 @@ interface IManageMealUseCase {
     suspend fun getAllMeals():List<Meal>
 }
 
-class ManageMealUseCase(private val remoteGateWay: IRemoteGateWay) : IManageMealUseCase {
+class ManageMealUseCase(private val remoteGateWay: FakeRemoteGateWay) : IManageMealUseCase {
```

Comment:
To utilize the correct implementation, you need to inject the interface and let the DI handle it.
```suggestion
class ManageMealUseCase(private val remoteGateWay: IRemoteGateWay) : IManageMealUseCase {
```